src/legacy/core_plugins/kibana/public/visualize/wizard/_dialog.scss

```diff
@@ -7,6 +7,7 @@
 }
 
 .visNewVisSearchDialog {
+  min-width: $euiSizeL * 30;
```

Comment:
Change this to `width` instead of `min-width` and then you shouldn't need the responsive override below.